### DISCUSSION TITLE
READ FIRST: Welcome to Visual Studio Code's Extension Community

### DISCUSSION THREAD

**Discussion by digitarald** [microsoft/vscode-discussions#5] [Announcements]
Community discussions is the place to discuss VS Code's extension platform, ask questions, help other members of the community, and get answers.

‼️ Discussions focus **only on extension development** to learn, experiment, and grow sustainably.

**[🙌  Ready to ask extension development questions? Start here!](https://github.com/microsoft/vscode-discussions/discussions)**

Other places to participate with VS Code:

- 🔎 `"Help: Report Issue"` is VS Code's built-in command to 🐛 report bugs, 🐌 performance issues, and 💡 feature requests.
- 🏠 [VS Code repo](https://github.com/microsoft/vscode) for triaging issues and upvoting [feature requests](https://go.microsoft.com/fwlink/?LinkID=533482), and [contributing](https://github.com/microsoft/vscode/wiki/How-to-Contribute).
- 📚 [StackOverflow](https://stackoverflow.com/questions/tagged/vscode) for general VS Code questions.
- 💬 [Extension Dev Slack](https://vscode-dev-community.slack.com/) for chatting with other extension authors.
- 🩻 [Extension API Documentation](https://code.visualstudio.com/api) and [code samples](https://github.com/microsoft/vscode-extension-samples) for learning and contributing documentation improvements.

### We need your participation

… to create a welcoming and inspiring community. [Microsoft's Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/) applies here, in short **please be kind and remember that there are real people on the other side of the keyboard.** 

Also:

- As we begin to build this community, we ask that you limit conversations to the categories available (more categories will be released as the community grows!)

- Before opening a new discussion, please search existing posts to see if the topic thread already exists.

- Please do not use this as a place to `@` your favorite VS Code contributor to get an extra bump on a feature request. Spoiler: it won't work.

### FAQ

**How can I help?**
Answer questions with empathy and kindness. Keep the community welcoming by giving contructive feedback and reporting unacceptable behavior. Reach out if you like to help with moderation.

**Why are there just categories for extension development?**
We want to start small, and the extension author community is just the right size. Meanwhile [StackOverflow](https://stackoverflow.com/questions/tagged/visual-studio-code?tab=Newest) continues to serve most other questions.

**When will more categories be added?**
As we learn what works well, tweak our moderation process, and see the community grow before expanding topics. If we see moderation becoming a burden on the team we will also consider take action.

**Comment by PEZ** [comment]
Such a wonderful initiative. Thanks for starting this forum!

**Reply by mikeb42069** [reply]
> HOLA A TODOS 
> 
You're welcome

**Comment by infinitepower18** [comment]
Is there any reason why it's on a separate repo rather than having the discussions on the main vscode repo?

**Reply by digitarald** [reply]
Mostly that we want to avoid mixing it up with the existing issue triage process (for now) and are considering adding more categories for topics that have their own repos, like remote, python, etc.

**Reply by Splines** [reply]
Just to let you know: it was the vs-code-engineering bot that just pointed me to this `vscode-discussions` repo. It might be worth to pin an issue on the main repo to let people know there exist these discussions to avoid that people ask questions in the usual issues (like I did since I didn't know about this)

**Reply by gjsjohnmurray** [reply]
@Splines please note this from above:

> ‼️ Discussions focus **only on extension development**

We already have a hard time redirecting people who come here with questions about **using** VS Code, and I reckon linking here from the main repo is likely to make that worse.

**Reply by Splines** [reply]
Yeah, I was referring only to extension development questions and thought it'd be useful to be directed here from the main repo when I have a specific question about extension development, e.g. [this one](https://github.com/microsoft/vscode/issues/232522) that I was now able to resolve with the help of the `vscode-discussions` I did not know about earlier. But of course just a proposal ;)

And the `vs-code-engineering` bot did already a great job in linking the discussions, so I'm not gonna' complain.

**Comment by gjsjohnmurray** [comment]
> Happy discussing! :)

@digitarald please add a link to https://github.com/microsoft/vscode-discussions/discussions at the end of that line. Then when I point people at https://github.com/microsoft/vscode-discussions/discussions/5 from elsewhere (e.g. an issue in microsoft/vscode) they can more easily get started with "happy discussing!".

**Comment by water-bai** [comment]
- VS Code Version: 1.72.2
- OS Version: macOS 12.1 Monterey(13-inch, M1, 2020)

### demo
[demo-for-issue](https://github.com/baikaishui01/demo-for-issue)

### my problem
it work normal when i am debugging.but when i publish, install and use command "issueDemo.showLint". it toast an error, like follow image
![image](https://user-images.githubusercontent.com/48601669/198847251-1c2cdcfa-7e1f-4202-a1c8-5519b988bc9b.png)


### it work normal when i am debugging. eg:
![image](https://user-images.githubusercontent.com/48601669/198846581-9d1d7747-d1f9-4226-8e0f-661c8d0a0a75.png)

**Reply by digitarald** [reply]
@baikaishui01 could you please file an issue for VS Code: https://github.com/microsoft/vscode/issues

**Reply by water-bai** [reply]
ok, thanks. This is very urgent for me. I hope you can reply as soon as possible. This is my issue: https://github.com/microsoft/vscode/issues/165028

**Comment by ttax00** [comment]
**`middleware` `provideDocumentSymbols` doesn't display children symbols**
Concerning piece of code. https://code.visualstudio.com/api/language-extensions/embedded-languages
Calling the command on `.html` virtual document. 
```ts
async provideDocumentSymbols(document, token, next) {
let result: undefined | DocumentSymbol[] = undefined;

result = await commands.executeCommand<DocumentSymbol[]>(
'vscode.executeDocumentSymbolProvider',
vdocUri(document),
);

return result;
}
```
this returns an array of `DocumentSymbol[]` which has `children: DocumentSymbol[]` properties. But vscode only display the top level symbols instead of everything. See screenshots here: https://github.com/TechTheAwesome/code-yew-server/issues/2

Edit: A quick fix right now is to manually unpack the childrens into a flat DocumentSymbol array. Also, on startup, `vscode.executeDocumentSymbolProvider` would return undefined. And requires a few calls before the correct parse would be returned.

**Reply by gjsjohnmurray** [reply]
@TechTheAwesome I recommend starting a new discussion thread rather than commenting on the "Welcome" thread.

**Reply by ttax00** [reply]
I was linked to here when trying to make an issue on vscode :sweat_smile: I'm not too familiar with the project layout, so you meant moving the issue here? https://github.com/microsoft/vscode-discussions/discussions

**Reply by gjsjohnmurray** [reply]
Correct. There's a link from this thread which is intended to take you there:

![image](https://user-images.githubusercontent.com/6726799/199680370-f1679065-7ed3-49a4-95e7-c45031b453c1.png)

**Reply by ttax00** [reply]
Thank you so much!

**Comment by sanjay51** [comment]
Hi - is there a way to disable vscode theming for webviews? 
Details: https://stackoverflow.com/questions/74357614/vscode-webview-remove-default-styles

Thanks.

**Reply by gjsjohnmurray** [reply]
Please open a fresh discussion thread rather than branching from this Welcome thread. See above if you aren't clear how to do that.

**Comment by ruiminwang** [comment]
Is there any SDK/API to pop up the changelog once our extension is updated in customer's VS Code?

**Reply by r3m0t** [reply]
It is against the UX guidelines, but you can store your version number in `ExtensionContext.globalState` and check if it's gone out of sync with your `ExtensionContext.extension.packageJSON.version`. Or it's better to use a version number in your source code that you only increment when there is a significant changelog entry

**Reply by r3m0t** [reply]
Oh, I might be misremembering - here's the guidelines https://code.visualstudio.com/api/ux-guidelines/notifications

**Comment by aayush420** [comment]
Is there a way to get the CommentThread given the range and the document? It would be useful to add comments programmatically, rather than using the UI element on the line number column.

**Reply by gjsjohnmurray** [reply]
> Is there a way to get the CommentThread given the range and the document? It would be useful to add comments programmatically, rather than using the UI element on the line number column.

@aayush420 please start a new discussion instead of adding your comment to this "Welcome" discussion.

**Reply by aayush420** [reply]
https://github.com/microsoft/vscode-discussions/discussions/336

**Comment by abhikp98** [comment]
I was using vs code to run Python codes. So I used a simple function to do some operations which have one argument. But inside function the argument's features are not getting listed or suggesting... 
Installed extensions are 
Python
Pylance
Autopep8
PyLint

**Reply by gjsjohnmurray** [reply]
@abhikp98 based on the information at the start of this thread, please seek assistance with your issue elsewhere. In your case the Pylance extension support channel might be a good place to begin.

**Reply by abhikp98** [reply]
I have already added Pylance.
For example:
def test(value):
    value.        # Here after putting dot, the features of string should come right? capitalise, count...

data = 'github' 
test(data)

**Reply by gjsjohnmurray** [reply]
As you are not developing an extension for VS Code this is not the correct place for your help request. Please see above.

**Comment by Priyanshukeshri** [comment]


**Reply by gjsjohnmurray** [reply]
@Priyanshukeshri I recommend starting a new discussion thread rather than subverting the "Welcome" thread.

**Comment by plrubbo** [comment]
Hello, anyway could help me setting up an expo in Visual Studio Code?

**Comment by CuriousDev289** [comment]
@digitarald - How can I get the line number when hover on variable while using Vs code debugger extension for external debugging?
 
I am developing a custom debugger extension , when I started working on on-hover variable while debugging any program, I found that, there is  no option to get line number to get the context along with value of the particular variable.

**Comment by xaxazak** [comment]
> **When will more categories be added?** As we learn what works well, tweak our moderation process, and see the community grow before expanding topics. If we see moderation becoming a burden on the team we will also consider take action.

Have you had enough time to learn what works well?
Will you consider adding more forums now, such as:

- vscode development (not extensions).
- vscode help (perhaps separate categories for "getting started+installation", "general usage").
- general (community, metadiscussion, software suggestions, etc).

**Reply by digitarald** [reply]
It's a lot to manage and we are a small team. We still need to see how well this community supports itself and helps each other.

**Comment by TPJelf** [comment]
Extension Dev slack link is dead.

**Reply by digitarald** [reply]
Fixed, should work again

**Comment by gjsjohnmurray** [comment]
@digitarald I'd like to suggest some edits to this issue which may help reduce the number of threads which get started here that are not related to **extension development**.

1.  I propose changing the title of this thread to "README FIRST before posting here". It's good that it is pinned, but I suspect some people aren't motivated to read something titled "Welcome to Visual Studio Code's Community Discussions".

2. Rephrase and restructure the first part of the post. My suggestion is:

> This is a place to discuss VS Code's **extension platform**, ask questions about extension development, help other extension developers, and get answers.

> We are initially scoping discussions to** just extension development** to learn, experiment, and grow sustainably.

> Other places to participate with VS Code:

... (as before)

> [🙌 Ready to ask **extension development** questions? Start here!](https://github.com/microsoft/vscode-discussions/discussions)

> We need your participation
... (as before)

3. Lock this thread (assuming GitHub Discussions supports locking).

**Reply by digitarald** [reply]
Done, great ideas – lets hope more people read the right parts. Thanks for helping to manage and grow this community!

**Comment by samestep** [comment]
@digitarald how can I join the Slack? I clicked the [link](https://vscode-dev-community.slack.com/) you provided, but it asks me to sign in with no option to sign up as far as I can see; I put in my email address and it simply sent me an email saying this:

> **Is this the right email?**
>
> It looks like there isn’t an account on VS Code Development Community tied to this email address.

**Reply by alefragnani** [reply]
Hi @samestep ,

Feel free to try this one, which I just created. 

Join me on Slack -- it’s a faster, simpler way to work. Sign up here, from any device: https://join.slack.com/t/vscode-dev-community/shared_invite/zt-344uf1bb8-T5iQGaSkZX_Q_WmEDHzzXw

It would be easier if Slack didn't require an invite, but it seems not possible today 😞 

Just be sure you already have a Slack account, otherwise you will have to sign up before, via their home page https://slack.com/signin#/signin.

**Reply by jeremiah-snee-openx** [reply]
@alefragnani ~~link it dead again. Can we get an updated link? Thanks~~

Actually found this (https://aka.ms/vscode-dev-community) that works on https://github.com/microsoft/vscode/blob/main/README.md

**Comment by KainosSLS** [comment]
Done.
